domain/consensus/processes/blockvalidator/proof_of_work_test.go

```diff
@@ -0,0 +1,81 @@
+package blockvalidator_test
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model/pow"
+	"github.com/kaspanet/kaspad/domain/consensus/ruleerrors"
+	"github.com/kaspanet/kaspad/util"
+
+	"math/rand"
+	"testing"
+
+	"github.com/kaspanet/kaspad/domain/consensus"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"github.com/pkg/errors"
+)
+
+// Set the flag "skip pow" to be false (second argument in the function)
+func TestPOW(t *testing.T) {
+	testutils.ForAllNets(t, false, func(t *testing.T, params *dagconfig.Params) {
+		// First, it checks the validation of the genesis's POW.
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, "TestPOW")
+		if err != nil {
+			t.Fatalf("Error setting up consensus: %+v", err)
+		}
+		defer teardown(false)
+		// Builds and checks block with invalid POW.
+		invalidBlockWrongPOW, _, err := tc.BuildBlockWithParents([]*externalapi.DomainHash{params.GenesisHash}, nil, nil)
+		invalidBlockWrongPOW = solveBlockWithWrongPOW(invalidBlockWrongPOW)
+		if err != nil {
```

Comment:
This condition should be one line higher

Reply:
fixed :)